src/Filters/molecules/FilterLayout.tsx

```diff
@@ -0,0 +1,324 @@
+      background-color: hsl(205, 100%, 72%);
+      cursor: pointer;
+    }
+  `}
+  ${({ isInnerContextButton }) => isInnerContextButton && css`
+    margin-right: 5px;
+  `}
+`;
+
+const ContentBox = styled.div<{ openState: IDropOpen, disabled: boolean, minWidth: number }>`
+  z-index: 100;
+  min-width: ${({ minWidth }) => minWidth}px;
+  position: absolute;
+  ${({ openState, disabled }) => openState && css`
+    display: ${openState.isOpen ? 'inline-block' : 'none'};
+    display: ${disabled && 'none'};
+    ${openState.position === 'bottom-right' && `
+      bottom: 0;
+      left: 0;
+      transform: translateY(calc(100% + 5px ));
+    `};
+    ${openState.position === 'bottom-left' && `
+      bottom: 0;
+      right: 0;
+      transform: translateY(calc(100% + 5px ));
+    `};
+    ${openState.position === 'top-left' && `
+      top: 0;
+      right: 0;
+      transform: translateY(calc( -100% - 5px ));
+    `};
+    ${openState.position === 'top-right' && `
+      top: 0;
+      left: 0;
+      transform: translateY(calc( -100% - 5px ));
+    `};
+  `};
+`;
+
+const ButtonWrapper = styled.div`
+  display: inline-block;
+`;
+
+const TopLine = styled.div`
+  ${({ theme }) => theme.styles.filters.dropdownContainer.topBorder};
+  height: 3px;
+  border-top-left-radius: 3px;
+  border-top-right-radius: 3px;
+`;
+
+const InnerBox = styled.div`
+  border-bottom-left-radius: 3px;
+  border-bottom-right-radius: 3px;
+  box-shadow: 0 4px 9px 0 hsla(204, 22%, 67%, 0.07);
+  border: solid 1px hsl(207,16%,86%);
+  border-top: none;
+  background-color: hsl(200, 23%, 97%);
+`;
+
+const LayoutGroup = styled.div`
+  display: flex;
+  justify-content: space-between;
+  padding: 8px 8px 0 10px;
+  width: 250px;
+`;
+
+const PaginationGroup = styled.div`
+  display: flex;
+  border-top: ${({ theme: { colors } }) => colors.divider} 1px solid;
+  padding: 8px 8px 0 10px;
+  align-items: center;
+  justify-content: space-between;
+  width: 250px;
+`;
+
+const IconWrapper = styled.div`
+  display: flex;
+`;
+
+const LayoutText = styled.p`
+  font-family: ${({ theme }) => theme.fontFamily.ui};
+  font-size: 14px;
+  color: #8f8f8f;
+`;
+
+const PaginationText = styled.p`
+  font-family: ${({ theme }) => theme.fontFamily.ui};
+  font-size: 14px;
+  color: #8f8f8f;
+`;
+
+const PageSizeContainer = styled.div`
+  display: flex;
+`;
+
+const SelectFieldContainer  = styled.div`
+  select{
+    background-color: #fff;
+  }
+`;
+
+interface IOptionsItem {
+  value: string
+  icon: string
+}
+
+interface IProps {
+  disabled?: boolean
+  onToggleOpenCallback?: (isOpen: boolean) => void
+  onCloseCallback?: () => void;
+  pageSizeOptions?: number[],
+  onPageSizeChange: (size: number) => void,
+  defaultPage?: number,
+  defaultPageSize: number,
+  getLayout: (layout: string) => void,
+  layoutText?: string,
+  pageSizeText?: string,
+  icon1: string,
+  icon2: string,
+  icon3?: string,
+  buttonValue1: string,
+  buttonValue2: string,
+  buttonValue3?: string
```

Comment:
I think it is better to have layoutData as attribute as array

layoutData =[
{
        icon: 'Status',
        id: 'grid1',
        tooltipText: 'Device ID:'
      },
{
        icon: 'Info',
        id: 'chart1',
        tooltipText: 'Bar Chart:'
      }
]

Also add another attribute hasPageSetting to show pagination dropdown in this layout.

Reply:
Done